apps/sms/views/conversation/js/conversation.js

```diff
@@ -2262,16 +2258,23 @@ var ConversationView = {
 
   // FIXME/bug 983411: phoneNumber not needed.
   simSelectedCallback: function conv_simSelected(phoneNumber, cardIndex) {
-    if (Compose.isEmpty()) {
-      return;
-    }
-
     cardIndex = +cardIndex;
+
     if (isNaN(cardIndex)) {
       cardIndex = 0;
     }
 
-    this.sendMessage({ serviceId: cardIndex });
+    var sendConfirmationPromise = Compose.isEmpty() ?
+      Utils.confirm(
+        'sendEmptyMessage-confirmation', null, { className: 'danger' }
+      ) :
+      Promise.resolve();
+
+    return sendConfirmationPromise.then(
+      () => this.sendMessage({ serviceId: cardIndex }), () => {
+        console.log('user doesn\'t confirm sending of empty messages');
```

Comment:
__nit:__ I wouldn't do logging here, it doesn't seem useful, moreover with this ```simSelectedCallback``` will always return ```resolved``` promise. Yes it will require an extra changes in unit tests (see below), but it's more logically correct IMO.

So since ```Promise.reject()``` of ```Utils.confirm``` doesn't contain any error we should do an extra check in unit tests
```
// When we expect promise to be resolved
clickButtonAndSelectSim().then(() => {
  ... asserts...
}, () => {
  throw new Error('Promise should not be rejected!');
}).then(done, done);
.....
// Otherwise
clickButtonAndSelectSim().then(() => {
  throw new Error('Promise should not be resolved!');
}, () => {
  ... asserts...
}).then(done, done);
```